internal/inventory/model_provenance.go

```
@@ -0,0 +1,762 @@
+	default:
+		return fmt.Errorf("unsupported model provenance confidence %q", provenance.Confidence)
+	}
+	if expected := modelDerivation(provenance.Distilled, provenance.Quantized); provenance.Derivation != expected {
+		return errors.New("model provenance derivation is inconsistent with quantized/distilled")
+	}
+	return nil
+}
+
+func modelBool(value bool) *bool { return &value }
+
+func cloneModelBool(value *bool) *bool {
```

golangci-lint (2.12.2)

[error] func cloneModelBool is unused

(unused)